<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>570</width>
    <height>400</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>570</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Ripple Offline</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="spacing">
     <number>1</number>
    </property>
    <property name="leftMargin">
     <number>9</number>
    </property>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>115</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>115</height>
       </size>
      </property>
      <property name="title">
       <string>Send XRP</string>
      </property>
      <widget class="QPushButton" name="buttonQR">
       <property name="geometry">
        <rect>
         <x>470</x>
         <y>25</y>
         <width>70</width>
         <height>70</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>70</width>
         <height>70</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>70</width>
         <height>70</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Generate Transaction</string>
       </property>
       <property name="layoutDirection">
        <enum>Qt::RightToLeft</enum>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="offline.qrc">
         <normaloff>:/res/qrSmall.png</normaloff>:/res/qrSmall.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
       <property name="shortcut">
        <string>Return</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="value">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>80</y>
         <width>150</width>
         <height>20</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Amount of XRP to send</string>
       </property>
       <property name="suffix">
        <string> XRP</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>999999999</number>
       </property>
      </widget>
      <widget class="QLabel" name="labelArrow">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>20</y>
         <width>150</width>
         <height>50</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>50</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap resource="offline.qrc">:/res/arrow.png</pixmap>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignHCenter|Qt::AlignTop</set>
       </property>
      </widget>
      <widget class="DropWidget" name="frameFrom">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>141</width>
         <height>80</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="acceptDrops">
        <bool>true</bool>
       </property>
       <property name="toolTip">
        <string>Drag&amp;Drop here account from which to send XRP. Only accounts with valid secret key could be used (yellow).</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <widget class="DropWidget" name="frameTo">
       <property name="geometry">
        <rect>
         <x>320</x>
         <y>20</y>
         <width>141</width>
         <height>80</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="acceptDrops">
        <bool>true</bool>
       </property>
       <property name="toolTip">
        <string>Drag&amp;Drop here account which will receive sent XRP.</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>121</width>
         <height>16</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel { color : gray; }</string>
       </property>
       <property name="text">
        <string>Sender</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="sequence">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>121</width>
         <height>20</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>The Sequence of next generated transaction. A transaction is only valid if the Sequence number is exactly 1 greater than the last-valided transaction from the same account.</string>
       </property>
       <property name="suffix">
        <string/>
       </property>
       <property name="prefix">
        <string>sequence </string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>999999999</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>330</x>
         <y>20</y>
         <width>121</width>
         <height>16</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel { color : gray; }</string>
       </property>
       <property name="text">
        <string>Recipient</string>
       </property>
      </widget>
      <widget class="QPushButton" name="seqHelp">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>79</y>
         <width>22</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>22</width>
         <height>22</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>22</width>
         <height>22</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Fixed</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>7</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Accounts</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolButton" name="buttonAdd">
          <property name="minimumSize">
           <size>
            <width>43</width>
            <height>43</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>43</width>
            <height>43</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="offline.qrc">
            <normaloff>:/res/plus.png</normaloff>:/res/plus.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>35</width>
            <height>35</height>
           </size>
          </property>
          <property name="popupMode">
           <enum>QToolButton::InstantPopup</enum>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonIconOnly</enum>
          </property>
          <property name="autoRaise">
           <bool>false</bool>
          </property>
          <property name="arrowType">
           <enum>Qt::NoArrow</enum>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="DeleteDropWidget" name="frameDelete">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>46</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>50</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <widget class="QLabel" name="labelDelete">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>46</width>
             <height>56</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>46</width>
             <height>56</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>46</width>
             <height>56</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Drop Account to Delete</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="pixmap">
            <pixmap resource="offline.qrc">:/res/bin.png</pixmap>
           </property>
           <property name="scaledContents">
            <bool>false</bool>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="WidgetList" name="accountList">
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="gridSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="viewMode">
         <enum>QListView::IconMode</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer_3">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Fixed</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>6</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer_4">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Minimum</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>4</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetMaximumSize</enum>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>www.xrpoffline.com  </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Donate here to support further development:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="WidgetList" name="donate">
          <property name="maximumSize">
           <size>
            <width>90</width>
            <height>18</height>
           </size>
          </property>
          <property name="sizeIncrement">
           <size>
            <width>0</width>
            <height>25</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Drag&amp;Drop to recipient area to generate donation transation</string>
          </property>
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">QListView { color : gray; }</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="viewMode">
           <enum>QListView::IconMode</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <action name="action_Open">
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="actionFs">
   <property name="text">
    <string>fs</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>DropWidget</class>
   <extends>QFrame</extends>
   <header>dragwidget.h</header>
   <container>1</container>
   <slots>
    <signal>itemChanged(DropWidget*)</signal>
    <slot>on_itemDeleted(int)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>DeleteDropWidget</class>
   <extends>QFrame</extends>
   <header>dragwidget.h</header>
   <container>1</container>
   <slots>
    <signal>itemDeleted(int)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>WidgetList</class>
   <extends>QListView</extends>
   <header>dragwidget.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>buttonQR</tabstop>
  <tabstop>accountList</tabstop>
  <tabstop>sequence</tabstop>
  <tabstop>value</tabstop>
 </tabstops>
 <resources>
  <include location="offline.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>frameDelete</sender>
   <signal>itemDeleted(int)</signal>
   <receiver>frameFrom</receiver>
   <slot>on_itemDeleted(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>428</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frameDelete</sender>
   <signal>itemDeleted(int)</signal>
   <receiver>frameTo</receiver>
   <slot>on_itemDeleted(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>428</y>
    </hint>
    <hint type="destinationlabel">
     <x>388</x>
     <y>96</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frameDelete</sender>
   <signal>itemDeleted(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_itemDeleted(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>428</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frameFrom</sender>
   <signal>itemChanged(DropWidget*)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_itemChanged(DropWidget*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>98</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frameTo</sender>
   <signal>itemChanged(DropWidget*)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_itemChanged(DropWidget*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>489</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>271</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_itemChanged(DropWidget*)</slot>
  <slot>on_itemDeleted(int)</slot>
  <slot>on_action_Open_triggered()</slot>
  <slot>on_action_Save_triggered()</slot>
 </slots>
</ui>
